guac_actix/src/network_endpoints.rs

```diff
@@ -0,0 +1,112 @@
+use actix_web::http::Method;
+use actix_web::*;
+
+use guac_core::channel_client::types::{Channel, UpdateTx};
+
+use guac_core::crypto::CryptoService;
+use guac_core::{CRYPTO, STORAGE};
+
+use guac_core::channel_client::ChannelManager;
+
+use failure::Error;
+
+use futures::Future;
+
+use NetworkRequest;
+
+use guac_core::counterparty::Counterparty;
+use std::net::SocketAddr;
+
+pub fn init_server(port: u16) {
+    server::new(|| {
+        App::new()
+            .resource("/update", |r| r.method(Method::POST).with(update_endpoint))
+            .resource("/propose", |r| {
+                r.method(Method::POST).with(propose_channel_endpoint)
+            })
+            .resource("/channel_created", |r| {
+                r.method(Method::POST).with(channel_created_endpoint)
+            })
+            .resource("/channel_joined", |r| {
+                r.method(Method::POST).with(channel_joined_endpoint)
+            })
+    }).bind(&format!("[::0]:{}", port))
+        .unwrap()
+        .start();
+}
+
+pub fn update_endpoint(
```

Comment:
does the borrow checker not let you return bare futures? I'm wondering why we have all these boxes and if we could simplify the type schemes of functions with a different structure.

Reply:
we can actually return impl Future, but I was having some issues with actix-web (which turned out to be another unrelated issue) that caused me to switch them to Box<Future<>> instead of impl Future. I'll switch them back